The i_filterdata is currently only used by the size-on-mds to store the
epoch number for the inode.  This could be moved to another field in
ldiskfs or elsewhere in the inode that isn't used by Lustre callers.

Index: linux-2.6.18-128.1.6/include/linux/fs.h
===================================================================
--- linux-2.6.18-128.1.6.orig/include/linux/fs.h	2009-04-14 21:05:43.000000000 -0600
+++ linux-2.6.18-128.1.6/include/linux/fs.h	2009-06-02 23:21:44.000000000 -0600
@@ -580,6 +580,7 @@
 		struct cdev		*i_cdev;
 	};
 	int			i_cindex;
+	void                    *i_filterdata;
 
 	__u32			i_generation;
 
